### PR TITLE
Add deployments related sidebar components

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshot from 2023-02-23 15-30-43
Screenshot from 2023-02-23 15-30-38
Screenshot from 2023-02-23 15-33-00
Screenshot from 2023-02-23 15-32-55
Screenshot from 2023-02-23 15-32-49

### CODE REVIEW

File: packages/shared/topology/factory/StyleNode.tsx [hunk 1]
```diff
@@ -271,6 +271,8 @@ const StyleNode: React.FunctionComponent<StyleNodeProps> = ({
             detailsLevel === ScaleDetailsLevel.high &&
             passedData.showStatusDecorator
           }
+          // eslint-disable-next-line @typescript-eslint/no-empty-function
+          onStatusDecoratorClick={() => {}}
```
Comment (alfonsomthd) [inline packages/shared/topology/factory/StyleNode.tsx:275]: Just an observation: you can avoid disabling the linter by returning null
```suggestion
          onStatusDecoratorClick={() => null}
```

File: packages/shared/topology/sidebar/deployments/DeploymentResources.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    if (usedMemory && _.isEmpty(errorUsedMemory) && !loadingUsedMemory) {
+      filteredPods[podIndex].metrics.memory = getPodMetric(
+        usedMemory.data.result,
+        filteredPods[podIndex].metadata.name
+      );
+    }
+  }
+
+  return (
+    <div className="co-m-pane__body topology-sidebar-tab-observe">
```
Comment (alfonsomthd) [inline packages/shared/topology/sidebar/deployments/DeploymentResources.tsx:153]: As I forgot to update this in my previous PR, I can take care of it in my StorageCluster tabs PR.
```suggestion
    <div className="odf-m-pane__body topology-sidebar-tab-observe">
```

**Comment by alfonsomthd** [comment]
Is this [file](packages/shared/topology/sidebar/deployments/index.ts) intentionally empty?

File: packages/shared/topology/sidebar/deployments/DeploymentObserve.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          utilizationQuery={utilizationQueries[DeploymentQueries.CPU_USAGE]}
+        />
+        <PrometheusUtilizationItem
+          title={t('Memory')}
+          humanizeValue={humanizeBinaryBytes}
+          utilizationQuery={utilizationQueries[DeploymentQueries.MEMORY_USAGE]}
+          byteDataType={ByteDataTypes.BinaryBytes}
+        />
+        <PrometheusUtilizationItem
+          title={t('Recieve Bandwidth')}
```
Comment (SanjalKatiyar) [inline packages/shared/topology/sidebar/deployments/DeploymentObserve.tsx:53]: nits:
```suggestion
          title={t('Recieve bandwidth')}
```

File: packages/shared/topology/sidebar/deployments/DeploymentResources.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    memoizedReplicaSets,
+    memoizedStatefulSets,
+    pods,
+    podsLoadError,
+    podsLoaded,
+    replicaSetsError,
+    replicaSetsLoaded,
+    statefulSetError,
+    statefulSetLoaded,
+  ]);
```
Comment (SanjalKatiyar) [inline packages/shared/topology/sidebar/deployments/DeploymentResources.tsx:120]: lesser variables for React to compare...
```suggestion
    const loaded =
      statefulSetLoaded && replicaSetsLoaded && daemonSetsLoaded && podsLoaded;
    const loadError =
      statefulSetError || replicaSetsError || daemonSetError || podsLoadError;
  const filteredPods = React.useMemo(() => {
    const resources = [
      ...memoizedReplicaSets,
      ...memoizedDaemonSets,
      ...memoizedStatefulSets,
      deployment,
    ];
    return loaded && !loadError && !_.isEmpty(pods)
      ? pods.filter((pod) => isPodInDeployment(pod, ...resources))
      : [];
  }, [
    loaded,
    loadError,
    deployment,
    memoizedDaemonSets,
    memoizedReplicaSets,
    memoizedStatefulSets,
    pods,
  ]);
```

File: packages/shared/topology/sidebar/deployments/DeploymentResources.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+  return (
+    <div className="odf-m-pane__body topology-sidebar-tab-observe">
+      <h3>
+        {t('Pods')} ({filteredPods.length})
+      </h3>
+      {filteredPods.length > 0 && (
+        <>
+          <PodsOverviewList pods={filteredPods}></PodsOverviewList>
+        </>
```
Comment (SanjalKatiyar) [inline packages/shared/topology/sidebar/deployments/DeploymentResources.tsx:160]: nits:
```suggestion
          <PodsOverviewList pods={filteredPods} />
```

File: packages/shared/queries/deployment.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import * as _ from 'lodash-es';
+
+export enum DeploymentQueries {
+  CPU_USAGE = 'CPU_USAGE',
+  MEMORY_USAGE = 'MEMORY_USAGE',
+  RECIEVE_BANDWIDTH = 'RECIEVE_BANDWIDTH',
```
Comment (alfonsomthd) [inline packages/shared/queries/deployment.ts:6]:
```suggestion
  RECEIVE_BANDWIDTH = 'RECEIVE_BANDWIDTH',
```

File: packages/shared/topology/sidebar/deployments/DeploymentObserve.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+          utilizationQuery={utilizationQueries[DeploymentQueries.CPU_USAGE]}
+        />
+        <PrometheusUtilizationItem
+          title={t('Memory')}
+          humanizeValue={humanizeBinaryBytes}
+          utilizationQuery={utilizationQueries[DeploymentQueries.MEMORY_USAGE]}
+          byteDataType={ByteDataTypes.BinaryBytes}
+        />
+        <PrometheusUtilizationItem
+          title={t('Recieve bandwidth')}
```
Comment (alfonsomthd) [inline packages/shared/topology/sidebar/deployments/DeploymentObserve.tsx:53]:
```suggestion
          title={t('Receive bandwidth')}
```

**Pull request merged**